py_zipkin/_encoding_helpers.py

```diff
@@ -1,12 +1,163 @@
 import socket
 from collections import namedtuple
 
+from enum import Enum
+
+from py_zipkin.exception import ZipkinError
+
+
 Endpoint = namedtuple(
     'Endpoint',
     ['service_name', 'ipv4', 'ipv6', 'port'],
 )
 
 
+_V1Span = namedtuple(
+    'V1Span',
+    ['trace_id', 'name', 'parent_id', 'id', 'timestamp', 'duration', 'endpoint',
+     'debug', 'annotations', 'binary_annotations', 'sa_endpoint'],
+)
+
+
+class Kind(Enum):
+    CLIENT = 'CLIENT'
+    SERVER = 'SERVER'
+    LOCAL = None
+
+
+_DROP_ANNOTATIONS_BY_KIND = {
+    Kind.CLIENT: {'ss', 'sr'},
+    Kind.SERVER: {'cs', 'cr'},
+}
+
+
+class Span(object):
```

Comment:
This separation of `Span` and `_V1Span` feels a little weird. Like the storage of the data is in `_V1Span`, but it isn't a subclass of `Span`, nor does it make sense for it to be.

Maybe it feels like a `SpanBuilder` rather than a `Span`. Or maybe it is just the name of the thing that feels strange. I dunno. I can't put my finger on it.

Reply:
yeah I didn't know how to model these different spans and went back and forth a few times between namedtuples and classes. I like the idea of SpanBuilder though since that's basically what that class is.